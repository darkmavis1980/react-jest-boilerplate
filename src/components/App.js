import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Home from './Home/Home';

import "../styles/main.scss";

const App = () => (
  <BrowserRouter>
    <div className="app">
      <div className="app-body">
        <div className="main-content">
          <Routes>
            <Route path="/" element={<Home />} />
          </Routes>
        </div>
      </div>
    </div>
  </BrowserRouter>
);

export default App;
