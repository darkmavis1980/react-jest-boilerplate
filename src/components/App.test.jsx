import React, { StrictMode } from 'react';
import TestRenderer from 'react-test-renderer';
import App from './App';

const wrapper = TestRenderer.create(<StrictMode><App /></StrictMode>);
const instance = wrapper.root;

describe('(Component) App', () => {
  it('should match the snapshot', () => {
    const tree = TestRenderer
      .create(<StrictMode><App /></StrictMode>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('should inject the Router component', () => {
    expect(instance.findByProps({className: 'app'}).children.length).toBe(1);
  });
});
