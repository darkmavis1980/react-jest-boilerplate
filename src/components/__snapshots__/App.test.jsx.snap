// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) App should match the snapshot 1`] = `
<div
  className="app"
>
  <div
    className="app-body"
  >
    <div
      className="main-content"
    >
      <div>
        <h1>
          Hello World
        </h1>
        <p>
          Loading...
        </p>
        <hr />
        <p>
          Loading...
        </p>
      </div>
    </div>
  </div>
</div>
`;
